dataset_conversion/convert_bids_to_nnUnetv2.py

```diff
@@ -50,10 +75,70 @@ def get_parser():
     parser.add_argument('--split', nargs='+', type=float, default=[0.8, 0.2],
                         help='Ratios of training (includes validation) and test splits lying between 0-1. '
                              'Example: --split 0.8 0.2')
-
     return parser
 
 
+def convert_subject(root, subject, contrast, label_suffix, path_out_images, path_out_labels, counter, list_images,
+                    list_labels, is_ses,session=None):
+    """Function to get image from original BIDS dataset modify if needed and place
+        it with a compatible name in nnUNet dataset.
+
+    Args:
+        root (str): Path to BIDS dataset directory .
+        subject (str): Subject name .
+        contrast (str): Type of contrast.
+        label_suffix (str): suffix of the label in derivatives .
+        path_out_images (str): path to the images directory in the new dataset (test or train).
+        path_out_labels (str): path to the labels directory in the new dataset (test or train).
+        counter (int): Number of subject already treated.
+        list_images (list): List with all the images path in the new dataset.
+        list_labels (list): List with all the labels path in the new dataset.
+        is_ses (bool): Is the dataset session structured.
+        session (str): Session name or None if dataset without session .
+
+    Returns:
+        list_images (list): List with all the images path in the new dataset with one new append.
+        list_labels (list): List with all the labels path in the new dataset with one new append.
+
+    """
+    if is_ses:
+        # TODO verify that using _ between subject and session is problematic or not
+        subject_image_file = os.path.join(root, subject, session, 'anat',
+                                          f"{subject}_{session}_{contrast}.nii.gz")
+        subject_label_file = os.path.join(root, 'derivatives', 'labels', subject, session, 'anat',
+                                          f"{subject}_{session}_{contrast}_{label_suffix}.nii.gz")
+        sub_name = str(Path(subject_image_file).name).split('_')[0] + '_' + \
+                   str(Path(subject_image_file).name).split('_')[1]
+    else:
+        subject_image_file = os.path.join(root, subject, 'anat',
+                                          f"{subject}_{contrast}.nii.gz")
+        subject_label_file = os.path.join(root, 'derivatives', 'labels', subject, 'anat',
+                                          f"{subject}_{contrast}_{label_suffix}.nii.gz")
```

Comment:
We recently concluded with @naga-karthik that image and label filenames should also include the taskname, as is done in the original v1 conversion script:

https://github.com/ivadomed/data-conversion/blob/90e1b5efa0700fb4288e1c182b4eb5836228aa85/dataset_conversion/convert_bids_to_nnunet.py#L196-L197

Reply:
Yes, having some form of the dataset name (note that `taskname` was changed to `dataset name` in V2 so when I refer to dataset name it is essentially referring to the task name) is useful as is shown in the [nnunet docs](https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/dataset_format.md). 

In [these lines](https://github.com/ivadomed/model_seg_sci/blob/2d32cd3d8cd556bc59ecd50bd4a6d88ba341ebda/dataset-conversion/convert_bids_to_nnUNetv2.py#L118-L119) from one the earliest commits, I had the dataset name while saving in the nnuent format

Reply:
ok so i add the dataset name before each subject name in the file name ?

Reply:
Yes!